crates/server/src/store/in_memory/inner.rs

```diff
@@ -526,6 +526,12 @@ impl InnerInMemory {
         let mut records: Vec<Record> = vec![];
 
         {
+            let finish_nsec_record =
```

Comment:
Why not a normal function? Either locally or (preferably IMO) at the module level.